src/core/public/chrome/ui/header/header_breadcrumbs.tsx

```diff
@@ -51,15 +51,27 @@ export function HeaderBreadcrumbs({ appTitle$, breadcrumbs$, breadcrumbsAppendEx
     ),
   }));
 
-  if (breadcrumbsAppendExtension) {
-    const lastCrumb = crumbs[crumbs.length - 1];
-    lastCrumb.text = (
-      <>
-        {lastCrumb.text}
-        <HeaderExtension extension={breadcrumbsAppendExtension.content} display={'inlineBlock'} />
-      </>
-    );
-  }
+  const Breadcrumbs = (
+    <EuiHeaderBreadcrumbs
+      breadcrumbs={crumbs}
+      max={10}
+      data-test-subj="breadcrumbs"
+      className={classNames({
+        'chrHeaderNavBreadcrumbs--withAppendExtension': !!breadcrumbsAppendExtension,
+      })}
+    />
+  );
+
+  if (!breadcrumbsAppendExtension) return Breadcrumbs;
 
-  return <EuiHeaderBreadcrumbs breadcrumbs={crumbs} max={10} data-test-subj="breadcrumbs" />;
+  return (
+    <EuiFlexGroup responsive={false} gutterSize={'none'} className="eui-textTruncate">
+      <EuiFlexItem grow={false} className="eui-textTruncate">
+        {Breadcrumbs}
+      </EuiFlexItem>
+      <EuiFlexItem grow={true} className="chrHeaderNavBreadcrumbs__appendExtensionContainer">
+        <HeaderExtension extension={breadcrumbsAppendExtension.content} />
+      </EuiFlexItem>
```

Comment:
NIT: do we really need `grow:true` on the extension flex? I think having both flex items set to `false` should work?

Reply:
Thanks, `false` is working 👍